main.cpp

```diff
@@ -181,6 +188,21 @@ int addFiles(const char* dirname, const char* subPath) {
                 continue;
             }
 
+            if (!s_addAllFiles) {
+                bool skip = false;
+                size_t ignored_file_names_count = sizeof(ignored_file_names) / sizeof(ignored_file_names[0]);
+                for (size_t i = 0; i < ignored_file_names_count; ++i) {
+                    if (strcmp(ent->d_name, ignored_file_names[i]) == 0) {
```

Comment:
maybe use strncmp? for files like .gitignore and such?

Reply:
I can add .gitignore and maybe .gitmodules into the excluded list, i think.

Reply:
either way will work I think :)

Reply:
or maybe we add option so the user can exclude whatever they want  (including wildcard) or would that be too much?

Reply:
It could be a nice feature, but i will leave the implementation as an exercise to anyone who actually needs it.

Reply:
Done!

Reply:
good enough for me :)